pkg/apis/app/v1alpha1/kogitoinfra_types.go

```diff
@@ -111,6 +117,14 @@ const (
 	FailureInfraConditionType KogitoInfraConditionType = "Failure"
 )
 
+// KogitoInfraVolume describes the data structure for volumes that should be mounted in the given service provided by this infra instance
+type KogitoInfraVolume struct {
+	// Mount is the Kubernetes VolumeMount referenced by this instance
+	Mount v1.VolumeMount `json:"mount"`
+	// NamedVolume describes the pod Volume reference
+	NamedVolume v1.Volume `json:"volume"`
+}
```

Comment:
We might need to review these attributes. The CRD is gigantic right now! I'll think about something tomorrow.

Reply:
Oh no, pointers won't help in this case. We need the types to be in the CRD for OpenAPI validation and integration with Kube. I came up with a `ConfigVolume` type that will handle only the volumes type that we need.